gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemTest.java

```diff
@@ -126,7 +153,11 @@ public void testConstructor() throws IOException {
         options.getCloudStorageOptions().toBuilder().setAppName("appName").build());
 
     // Verify that credentials == null works - this is required for unauthenticated access.
-    new GoogleCloudStorageFileSystemImpl((Credentials) null, optionsBuilder.build());
+    // TODO: this seems to be not the case with java-storage.
+    // Confirm and remove this `if`.
+    if (!testStorageClientImpl) {
+      new GoogleCloudStorageFileSystemImpl((Credentials) null, optionsBuilder.build());
```

Comment:
What is the purpose of this code?

Reply:
`null` credentials were not allowed with new java-storage client.

Although there is a way to send unauthorized request to new storage client will add code changes for it.